### PR TITLE
[PREVIEW] DO NOT MERGE! Rdm 3064 Case progression: Better component contract

### PR DESCRIPTION
**Pull request by andrewfolga**
DO NOT MERGE! ONLY FOR POC PURPOSE.

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-3064

### Change description ###
Case progression: Better component contract:
1. Introduce Create Case component
2. Introduce Progress Case component

This is raised to show how the case progression facade components can be used in a project.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/app.config.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export class AppConfig extends AbstractAppConfig {
         .catch((error: any): any => {
           console.error(`Configuration ${configUrl} could not be read`, error);
           reject();
-          return Observable.throw(error.json().error || 'Server error');
+          return throwError(error.json().error || 'Server error');
```
Comment (vlaurin) [inline src/app/app.config.ts]: That change is not related to the POC and is a good change to bring in the code base

File: src/app/workbasket/workbasket.component.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export class WorkbasketComponent implements OnInit {
     let paginationMetadataObservable = this.paginationService
       .getPaginationMetadata(filter.jurisdiction.id, filter.caseType.id, metadataPaginationParams, caseFilters);
 
-    Observable.forkJoin(searchObservable, paginationMetadataObservable)
+    forkJoin(searchObservable, paginationMetadataObservable)
```
Comment (vlaurin) [inline src/app/workbasket/workbasket.component.ts]: That change is not related to the POC and is a good change to bring in the code base

File: src/app/cases/printer/case-printer.component.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { CasePrintDocument } from '../../shared/domain/case-view/case-print-document.model';
-import { CaseView } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseView, CasePrintDocument } from '@hmcts/ccd-case-ui-toolkit';
```
Comment (vlaurin) [inline src/app/cases/printer/case-printer.component.ts]: Aren't those changes needed in the codebase as well?

File: src/app/app.config.ts (hunk 1 above)
Comment (andrewfolga) [reply src/app/app.config.ts]: Sure will bring it in.

File: src/app/workbasket/workbasket.component.ts (hunk 2 above)
Comment (andrewfolga) [reply src/app/workbasket/workbasket.component.ts]: Sure will bring it in.

File: src/app/cases/printer/case-printer.component.ts (hunk 3 above)
Comment (andrewfolga) [reply src/app/cases/printer/case-printer.component.ts]: yep will bring it in

**Pull request closed without merging**